client/daemon/peer/piece_dispatcher.go

```diff
@@ -0,0 +1,157 @@
+/*
+ *     Copyright 2022 The Dragonfly Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package peer
+
+import (
+	"errors"
+	"math/rand"
+	"sync"
+	"time"
+
+	"go.uber.org/atomic"
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+
+	logger "d7y.io/dragonfly/v2/internal/dflog"
+)
+
+type PieceDispatcher interface {
+	Put(req *DownloadPieceRequest)
+	Get() (req *DownloadPieceRequest, err error)
+	Report(result *DownloadPieceResult)
+	Close()
+}
+
+var ErrNoValidPieceTemporarily = errors.New("no valid piece temporarily")
+
+type pieceDispatcher struct {
+	reqMap     map[string][]*DownloadPieceRequest
```

Comment:
Naming does not require a map type.

Reply:
ReqMap is not a naming map. It's key is peerID, value is a slice of piece requests.